<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        <link rel="canonical" href="https://sourmash-bio.github.io/sourmash-examples/15-estimate-how-much-of-a-metagenome-sample-can-be-mapped-to-reference-genomes.html">
        <link rel="shortcut icon" href="img/favicon.ico">
        <title>Example: estimate how much of a metagenome sample can be mapped to reference genomes - sourmash examples</title>
        <link href="css/bootstrap.min.css" rel="stylesheet">
        <link href="css/font-awesome.min.css" rel="stylesheet">
        <link href="css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css">

        <script src="js/jquery-1.10.2.min.js" defer></script>
        <script src="js/bootstrap.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script> 
    </head>

    <body>
        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-primary">
            <div class="container">
                <a class="navbar-brand" href="index.html">sourmash examples</a>
                <!-- Expander button -->
                <button type="button" class="navbar-toggler" data-toggle="collapse" data-target="#navbar-collapse">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <!-- Expanded navigation -->
                <div id="navbar-collapse" class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="navitem">
                                <a href="index.html" class="nav-link">Home</a>
                            </li>
                            <li class="navitem">
                                <a href="examples.html" class="nav-link">All examples</a>
                            </li>
                            <li class="navitem">
                                <a href="labels.html" class="nav-link">All categories</a>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav ml-auto">
                        <li class="nav-item">
                            <a href="#" class="nav-link" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="row">
                    <div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
    <div class="navbar-header">
        <button type="button" class="navbar-toggler collapsed" data-toggle="collapse" data-target="#toc-collapse" title="Table of Contents">
            <span class="fa fa-angle-down"></span>
        </button>
    </div>

    
    <div id="toc-collapse" class="navbar-collapse collapse card bg-secondary">
        <ul class="nav flex-column">
            
            <li class="nav-item" data-level="1"><a href="#example-estimate-how-much-of-a-metagenome-sample-can-be-mapped-to-reference-genomes" class="nav-link">Example: estimate how much of a metagenome sample can be mapped to reference genomes</a>
              <ul class="nav flex-column">
            <li class="nav-item" data-level="2"><a href="#caveats-and-details" class="nav-link">Caveats and details</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#publications-and-software" class="nav-link">Publications and software</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#version-notes" class="nav-link">Version notes</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#categories" class="nav-link">Categories</a>
              <ul class="nav flex-column">
              </ul>
            </li>
              </ul>
            </li>
        </ul>
    </div>
</div></div>
                    <div class="col-md-9" role="main">

<h1 id="example-estimate-how-much-of-a-metagenome-sample-can-be-mapped-to-reference-genomes">Example: estimate how much of a metagenome sample can be mapped to reference genomes</h1>
<p><em><a href="https://github.com/sourmash-bio/sourmash-examples/issues/15">sourmash-bio/sourmash-examples#15</a></em></p>
<hr />
<p>One question we frequently have about a metagenome is how much of it is <em>known</em>, and one way this can be answered is by asking <strong>how many of my metagenome reads can be mapped to a set of reference genome(s)?</strong></p>
<p>This can be estimated with sourmash fairly quickly, using any collection of reference genomes you have available!</p>
<p>In <a href="14-analyze-a-metagenome-using-66000-gtdb-genomic-representatives.html">Example: analyze a metagenome using 66,000 GTDB genomic representatives</a>, we showed how to analyze a metagenome's contents using <code>sourmash gather</code> against the GTDB genomic representatives database. The output was:</p>
<pre><code>overlap     p_query p_match avg_abund
---------   ------- ------- ---------
383.0 kbp      2.0%    7.8%       1.6    GCF_003697165.2 Escherichia coli DSM ...
187.0 kbp      1.0%    5.0%       1.6    GCF_015074785.1 Prevotella copri stra...
142.0 kbp      0.7%    2.8%       1.4    GCF_000012825.1 Bacteroides vulgatus ...
164.0 kbp      0.3%    1.4%       1.7    GCF_019127135.1 Prevotella copri stra...
found less than 50.0 kbp in common. =&gt; exiting

found 4 matches total;
the recovered matches hit 4.0% of the abundance-weighted query
</code></pre>
<p>and it turns out that these numbers provide accurate estimates for mapping rates.</p>
<p>Specifically:</p>
<p>the <code>p_query</code> column tells you <strong>what fraction of the metagenome reads will map to each genome.</strong> So here about 2% of the reads will map to  <code>GCF_003697165.2</code>, an E. coli genome, and about 1% of the reads will map to <code>GCF_015074785.1</code>, a Prevotella genome.</p>
<p>The <code>p_match</code> column tells you <strong>what fraction of the bases in the genome will be covered by the mapped reads</strong>. So for the <code>GCF_003697165.2</code> E. coli genome, about 7.8% of the genome will be covered by reads from this metagenome.</p>
<p>Finally, the statement "the recovered matches hit 4.0% of the abundance-weighted query" tells you that <strong>overall, 4% of the reads in the metagenome</strong> will map to your database.**</p>
<p>All of these numbers (and many more) are also available in the CSV output. See <a href="https://sourmash.readthedocs.io/en/latest/classifying-signatures.html#appendix-a-how-sourmash-gather-works">this section of the sourmash docs</a> for some details on how <code>sourmash gather</code> works, what is output to a CSV file, and  how the CSV output</p>
<h2 id="caveats-and-details">Caveats and details</h2>
<p>A few terms and conditions apply -</p>
<ul>
<li>you need to use abundance-weighted signatures, as calculated here: <a href="12-create-a-metagenome-signature-from-shotgun-sequencing-reads.html">Example: create a metagenome signature from shotgun sequencing reads</a></li>
<li>the fraction of the genome that can be classified against a database very much depends on that database! So you  want to use the most inclusive database that meets your needs. This can include both <a href="https://sourmash.readthedocs.io/en/latest/databases.html">prepared databases</a> like Genbank, and private databases (see <a href="1-download-sketch-and-search-a-collection-of-fasta-files.html">Example: download, sketch, and search a collection of FASTA files</a>).</li>
</ul>
<h2 id="publications-and-software">Publications and software</h2>
<p>If you're curious about the scientific details and want to see this applied to some larger metagenomes, please see <a href="https://www.biorxiv.org/content/10.1101/2022.01.11.475838v2">Lightweight compositional analysis of metagenomes with FracMinHash and minimum metagenome covers, Irber et al., 2022</a>.</p>
<p>Last but not least, the <a href="https://dib-lab.github.io/genome-grist/">genome-grist software</a> software provides a workflow that will map Illumina metagenome reads to a collection of reference genomes.</p>
<h2 id="version-notes">Version notes</h2>
<p>The summary output statement (<code>the recovered matches hit 4.0% of the abundance-weighted query</code>) was introduced in sourmash v4.2.4, with PR <a href="https://github.com/sourmash-bio/sourmash/pull/1819">sourmash-bio/sourmash#1819</a>. Prior to that, sourmash didn't distinguish between abundance-weighted and flat output in its text output (although the numbers were weighted properly - it just didn't tell you!)</p>
<p>In the future we plan to output both flat and abundance-weighted numbers; see <a href="https://github.com/sourmash-bio/sourmash/issues/1818">sourmash-bio/sourmash#1818</a>.</p>
<h2 id="categories">Categories</h2>
<p>This example belongs to the following categories:</p>
<ul>
<li><a href="l-sourmash-v4.2.html">examples using sourmash v4.2 functionality</a></li>
<li><a href="l-gtdb-rs207.html">examples using GTDB RS207</a></li>
<li><a href="l-intermediate.html">intermediate examples</a></li>
<li><a href="l-metagenome.html">analyzing metagenomes</a></li>
<li><a href="l-mapping.html">examples that refer to read mapping</a></li>
</ul></div>
            </div>
        </div>

        <footer class="col-md-12">
            <hr>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = ".",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="js/base.js" defer></script>
        <script src="search/main.js" defer></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="searchModalLabel" aria-hidden="true">
    <div class="modal-dialog modal-lg">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="searchModalLabel">Search</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
                <p>From here you can search these documents. Enter your search terms below.</p>
                <form>
                    <div class="form-group">
                        <input type="search" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results" data-no-results-text="No results found"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="keyboardModalLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
