<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        <link rel="canonical" href="https://sourmash-bio.github.io/sourmash-examples/16-compare-and-visualize-the-similarity-of-many-genomes-using-average-nucleotide-identity-ani.html">
        <link rel="shortcut icon" href="img/favicon.ico">
        <title>Example: compare and visualize the similarity of many genomes using Average Nucleotide Identity (ANI) - sourmash examples</title>
        <link href="css/bootstrap.min.css" rel="stylesheet">
        <link href="css/font-awesome.min.css" rel="stylesheet">
        <link href="css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css">

        <script src="js/jquery-1.10.2.min.js" defer></script>
        <script src="js/bootstrap.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script> 
    </head>

    <body>
        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-primary">
            <div class="container">
                <a class="navbar-brand" href="index.html">sourmash examples</a>
                <!-- Expander button -->
                <button type="button" class="navbar-toggler" data-toggle="collapse" data-target="#navbar-collapse">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <!-- Expanded navigation -->
                <div id="navbar-collapse" class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="navitem">
                                <a href="index.html" class="nav-link">Home</a>
                            </li>
                            <li class="navitem">
                                <a href="examples.html" class="nav-link">All examples</a>
                            </li>
                            <li class="navitem">
                                <a href="labels.html" class="nav-link">All categories</a>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav ml-auto">
                        <li class="nav-item">
                            <a href="#" class="nav-link" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="row">
                    <div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
    <div class="navbar-header">
        <button type="button" class="navbar-toggler collapsed" data-toggle="collapse" data-target="#toc-collapse" title="Table of Contents">
            <span class="fa fa-angle-down"></span>
        </button>
    </div>

    
    <div id="toc-collapse" class="navbar-collapse collapse card bg-secondary">
        <ul class="nav flex-column">
            
            <li class="nav-item" data-level="1"><a href="#example-compare-and-visualize-the-similarity-of-many-genomes-using-average-nucleotide-identity-ani" class="nav-link">Example: compare and visualize the similarity of many genomes using Average Nucleotide Identity (ANI)</a>
              <ul class="nav flex-column">
            <li class="nav-item" data-level="2"><a href="#caveats-and-details" class="nav-link">Caveats and details</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#citations" class="nav-link">Citations</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#categories" class="nav-link">Categories</a>
              <ul class="nav flex-column">
              </ul>
            </li>
              </ul>
            </li>
        </ul>
    </div>
</div></div>
                    <div class="col-md-9" role="main">

<h1 id="example-compare-and-visualize-the-similarity-of-many-genomes-using-average-nucleotide-identity-ani">Example: compare and visualize the similarity of many genomes using Average Nucleotide Identity (ANI)</h1>
<p><em><a href="https://github.com/sourmash-bio/sourmash-examples/issues/16">sourmash-bio/sourmash-examples#16</a></em></p>
<hr />
<p>with sourmash v4.4, sourmash can now estimate Average Nucleotide Identity between genomes: this is the fraction of bases that would be identical in a pairwise alignment. However, sourmash estimates this based on k-mers, instead of using alignments. This is fast and lightweight, and doesn't need access to the full genomes!</p>
<p>Let's use ANI with <code>sourmash compare</code> and <code>sourmash plot</code> to look at the relationship between 12 <em>E. coli</em> genomes.</p>
<p>First, you'll need to download the GTDB genomic-reps database, containing 66k genomes, as in <a href="13-download-and-search-66000-gtdb-genomes-with-a-query-genome.html">Example: download and search 66,000 GTDB genomes with a query genome</a>.</p>
<p>Then, run <code>compare</code>, selecting just the <code>Escherichia</code> genomes:</p>
<pre><code class="language-shell">sourmash compare --include Escherichia gtdb-rs207.genomic-reps.dna.k31.zip --ani -o ecoli-ani.cmp
</code></pre>
<p>This creates a numpy comparison matrix in <code>ecoli-ani.cmp</code> (you can also generate a CSV output with <code>--csv</code>).</p>
<p>Now, use <code>plot</code> to quickly visualize the comparison matrix:</p>
<pre><code class="language-shell">sourmash plot ecoli-ani.cmp
</code></pre>
<p>This will produce an image <code>ecoli-ani.cmp/matrix.png</code>:</p>
<p><img alt="ecoli-ani cmp matrix" src="https://user-images.githubusercontent.com/51016/168434572-0a358eea-ee64-4cbd-a1ad-1cfbeeba7846.png" /></p>
<p>This plot shows the clade structure of the 12 <code>E. coli</code> genomes, including three that have no detectable ANI similarity to any other ANI genomes.</p>
<p>The genome names corresponding to the label numbers can be found in the <code>ecoli-ani.cmp.labels.txt</code> file.</p>
<h2 id="caveats-and-details">Caveats and details</h2>
<p>ANI can be estimated by sourmash from ~85% to 99%, depending on the k-mer size used. See <a href="https://github.com/sourmash-bio/sourmash/issues/1859">https://github.com/sourmash-bio/sourmash/issues/1859</a> for some numbers for k=21 and k=31.</p>
<p>ANI estimates are somewhat dependent on the software and parameters used to calculate them. We are working on a systematic comparison of sourmash's ANI estimates with other ANI software!</p>
<p>sourmash ANI estimates are only available for scaled signatures (the default, when signatures are generated with <code>sourmash sketch</code>).</p>
<p>In order to accurately estimate ANI, sourmash signatures need to have enough hashes for the calculation; this is dependent on both the size of the genome(s) and the scaled factor used to generate the signatures. sourmash will output warnings to stderr when the sketches are too small to accurately estimate ANI.</p>
<p>Some details on how sourmash estimates ANI are <a href="https://sourmash.readthedocs.io/en/latest/classifying-signatures.html?highlight=ani#estimating-ani-from-fracminhash-comparisons">in the sourmash docs</a>.</p>
<h2 id="citations">Citations</h2>
<p>The analytical work underlying the ANI calculations is introduced in <a href="https://www.biorxiv.org/content/10.1101/2022.01.11.475870v2">Debiasing FracMinHash and deriving confidence intervals for mutation rates across a wide range of evolutionary distances, Hera et al., 2022</a>.</p>
<h2 id="categories">Categories</h2>
<p>This example belongs to the following categories:</p>
<ul>
<li><a href="l-sourmash-v4.4.html">examples using sourmash v4.4 functionality</a></li>
<li><a href="l-ani.html">using Average Nucleotide Identity (ANI)</a></li>
<li><a href="l-plotting.html">plots and other output visualizations</a></li>
</ul></div>
            </div>
        </div>

        <footer class="col-md-12">
            <hr>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = ".",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="js/base.js" defer></script>
        <script src="search/main.js" defer></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="searchModalLabel" aria-hidden="true">
    <div class="modal-dialog modal-lg">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="searchModalLabel">Search</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
                <p>From here you can search these documents. Enter your search terms below.</p>
                <form>
                    <div class="form-group">
                        <input type="search" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results" data-no-results-text="No results found"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="keyboardModalLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
